### PR TITLE
roachtest: fix sqlalchemy roachtest after alembic support

### PR DESCRIPTION
**Pull request by arulajmani**
Previously, we were running the sqlalchemy test suite from the
sqlalchemy repo. This patch changes it to trigger the test suite from
cockroachdb-sqlalchemy repo (our dialect). Running it from the
sqlalchemy repo was broken after adding alembic support.

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/sqlalchemy.go
Comment (rafiss) [inline pkg/cmd/roachtest/sqlalchemy.go:123]: i feel like this should be pinned to a specific commit, otherwise if alembic changes upstream, this test could start failing suddently

File: pkg/cmd/roachtest/sqlalchemy.go
Comment (rafiss) [inline pkg/cmd/roachtest/sqlalchemy.go:181]: nice, so this aligns more with how the sqlalchemy-cockroachdb CI works? lgtm as long as @gordthompson can confirm

**Review by rafiss — CHANGES_REQUESTED** [review]
just one comment. thank you!

File: pkg/cmd/roachtest/sqlalchemy.go
Comment (gordthompson) [reply pkg/cmd/roachtest/sqlalchemy.go:123]: If a change gets merged into alembic/master then it has already passed all of the Alembic CI tests, including testing against the five internal SQLA dialects (mssql, mysql, oracle, postgresql, and sqlite). One could argue that it's better to test against latest master so if something does start failing we can more quickly identify if it is an issue with Alembic, or with the dialect, or the database itself. It's your call.

File: pkg/cmd/roachtest/sqlalchemy.go
Comment (gordthompson) [reply pkg/cmd/roachtest/sqlalchemy.go:181]: Note that running

pytest … test/test_suite_sqlalchemy.py

will only run the SQLAlchemy tests, excluding the Alembic tests and your own specific tests (test_introspection.py, etc.). To run all of the tests just do

pytest --maxfail=0 --dburi=cockroachdb://root@localhost:26257/defaultdb?sslmode=disable

File: pkg/cmd/roachtest/sqlalchemy.go
Comment (rafiss) [reply pkg/cmd/roachtest/sqlalchemy.go:123]: yeah, we've experienced the benefits (and pains) of testing against the latest `master` version in the past, which is why the other projects are pinned to a specific version now. we found that it caused more maintenance burden than it was worth, since our nightly tests could start failing without us taking any action. we also wouldn't have time to debug right away, making it harder to debug failures that started weeks in the past. totally agree that pinning comes with the drawback of not testing against newer changes, but we've set a periodic task for our team to update the versions of all these tests.

so i vote for leaving it pinned for now.

File: pkg/cmd/roachtest/sqlalchemy.go
Comment (arulajmani) [reply pkg/cmd/roachtest/sqlalchemy.go:181]: That makes sense, I have it on my radar to do that as a followup, as it could require updating some blocklists that we maintain on our end.

**Review by arulajmani — COMMENTED** [review]
I pinned the alembic version, RFAL

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66918#-:-MdJNON31Lpr2-GAQijf:b-hfw65j)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gordthompson and @rafiss)

**Review by gordthompson — APPROVED** [review]
Looks good to me.

**Comment by arulajmani** [comment]
@rafiss wanna stamp this as well before I bors?

**Review by rafiss — APPROVED** [review]
looks good!

**Comment by rafiss** [comment]
@arulajmani can you backport this? see https://github.com/cockroachdb/cockroach/issues/66756

**Pull request merged**
